lib/auth/auth.go

```diff
@@ -2018,9 +2018,9 @@ func (a *Server) CreateWebSession(ctx context.Context, user string) (types.WebSe
 func (a *Server) GenerateToken(ctx context.Context, req *proto.GenerateTokenRequest) (string, error) {
 	expires := a.clock.Now().UTC()
 	if req.TTL != 0 {
-		expires.Add(req.TTL.Get())
+		expires = expires.Add(req.TTL.Get())
 	} else {
-		expires.Add(defaults.ProvisioningTokenTTL)
+		expires = expires.Add(defaults.ProvisioningTokenTTL)
 	}
```

Comment:
```go
tokenTTL:=defaults.ProvisioningTokenTTL
if req.TTL != 0 {
tokenTTL= req.TTL
}
expires := a.clock.Now().UTC().Add( tokenTTL)
```

Seems easier to follow

Reply:
Agreed.

Reply:
Addressed in [3e224d5](https://github.com/gravitational/teleport/pull/14936/commits/3e224d5d8361853b7981db505116931135d2637d)